🐛 Bug: Published declarations break with @types/estree@1.0.9 under strict typecheck (no skipLibCheck)

### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have [searched for related issues](https://github.com/michaelfaith/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Importing the plugin in a TypeScript ESLint config (`eslint.config.ts`) should typecheck cleanly when the project resolves `@types/estree@1.0.9`, without needing `skipLibCheck`.

### Actual

`tsc --noEmit` on a strict project (no `skipLibCheck`) that imports `eslint-plugin-package-json` fails with three TS2430 errors inside the published declarations:

```
node_modules/eslint-plugin-package-json/dist/createRule-B1Ow8h0I.d.mts(507,11): error TS2430: Interface 'PackageAST' incorrectly extends interface 'Program'.
  Types of property 'body' are incompatible.
    Type '[ASTBodyStatement]' is not assignable to type '(ModuleDeclaration | Statement | Directive)[]'.
      ...
        Property 'directive' is missing in type 'ASTBodyStatement' but required in type 'Directive'.
node_modules/eslint-plugin-package-json/dist/createRule-B1Ow8h0I.d.mts(519,11): error TS2430: Interface 'PackageJsonRuleContext<Options>' incorrectly extends interface 'RuleContext'.
node_modules/eslint-plugin-package-json/dist/createRule-B1Ow8h0I.d.mts(537,11): error TS2430: Interface 'PackageJsonSourceCode' incorrectly extends interface 'SourceCode'.
```

### Root cause

The bundled `.d.mts` inlines a full copy of the estree node types **as they existed in `@types/estree@1.0.8`** (e.g. `Property.key: Expression | PrivateIdentifier` — the repo pins `"@types/estree": "1.0.8"` in devDependencies), while `PackageAST extends Program` references the **installed** `@types/estree` through the `eslint` types.

`@types/estree@1.0.9` (published 2026-05-06) narrowed `Property.key` from `Expression | PrivateIdentifier` to `Expression` — that is the only change between 1.0.8 and 1.0.9:

```diff
 export interface Property extends BaseNode {
     type: "Property";
-    key: Expression | PrivateIdentifier;
+    key: Expression;
```

Since current `eslint` resolves `@types/estree` to 1.0.9, the inlined 1.0.8 shapes are no longer mutually assignable with the installed 1.0.9 shapes, so the `extends` clauses fail. Pinning `@types/estree` to 1.0.8 via package-manager overrides makes the errors disappear, which confirms the version skew.

### Version

1.6.0 (also reproduces on 1.0.0 and 1.5.0 — the inlining predates them)

### ESLint Version

10.7.0

### Additional Info

Environment: TypeScript 6.0.3, `strict: true`, no `skipLibCheck`, `moduleResolution: bundler`.

Two possible fixes, in order of durability:

1. Reference `@types/estree` as a real (runtime-irrelevant) dependency instead of inlining a copy into the bundled declarations — then the consumer's single installed version is used everywhere and no skew is possible.
2. Bump the `@types/estree` devDependency to `^1.0.9` and republish, so the inlined copy matches what current `eslint` resolves. (This keeps the inlining approach, so the same class of breakage can recur whenever `@types/estree` changes shape.)

Happy to send a PR for either direction if you let me know which you'd prefer.

Our current downstream workaround, for anyone else hitting this: a local `declare module 'eslint-plugin-package-json'` override exposing just `configs`, wired via `paths` in tsconfig.

--- a/package.json
+++ b/package.json
@@ -118,10 +118,14 @@
   },
   "peerDependencies": {
     "@eslint/json": ">=1.0.0",
+    "@types/estree": ">=1.0.0",
     "eslint": ">=9.0.0"
   },
   "peerDependenciesMeta": {
     "@eslint/json": {
+      "optional": true
+    },
+    "@types/estree": {
       "optional": true
     }
   },

